### ISSUE TITLE
Is there an equivalent way to achieve Node's readable.unshift method using Deno's ReadableStream?

### ISSUE THREAD

**Issue by onegosmer** [denoland/deno#17000]
I want to port some code written for Node to Deno, but I ran into an issue with Deno's [ReadableStream](https://deno.land/api@v1.28.3?s=ReadableStream) lacking an unshift method, like what Node has with [readable.unshift](https://nodejs.org/api/stream.html#readableunshiftchunk-encoding).

I know there is Node Compatibility Mode in Deno, but the Deno ReadableStream I have to use is coming from [this Deno API](https://deno.land/api@v1.28.3?s=Deno.Conn) so I cannot substitute it for the Node Compatibility Mode API.

Is there a way to emulate the behavior of Node's unshift method using a Deno ReadableStream, so data can be put back onto the Deno stream?

**Comment by crowlKats** [comment]
The Web Streams API does not have an equivalent of it

**Comment by onegosmer** [comment]
Ah, got it. I was able to work around the issue by storing the buffer that would need to be written using the missing unshift method, then combining it with the buffer from the next read of the stream, which functions similarly to what unshift does without directly involving the stream itself. Thank you for your response.

**Comment by crowlKats** [comment]
Also, the node docs state:
> Developers using stream.unshift() often should consider switching to use of a [Transform](https://nodejs.org/api/stream.html#class-streamtransform) stream instead.

of which there is an equivalent (`TransformStream`)

**Comment by onegosmer** [comment]
Yeah, I considered that but was unsure how to update the legacy Node code that I need to work with since it assumes everything comes from the single source stream, which gets read all over the place. It might be helpful though. Thank you for pointing it out.
